bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttException.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.transport.mqtt;
+
+/**
+ * Thrown if an error occurs communicating with the server.
+ * The exception contains a reason code. The semantic of the reason
+ * code depends on the underlying implementation.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class MqttException extends Exception {
+    private static final long serialVersionUID = 301L;
+    private int reasonCode;
+    private Throwable cause;
+
+    /**
+     * Constructs a new <code>MqttException</code> with the specified code
+     * as the underlying reason.
+     *
+     * @param reasonCode the reason code for the exception.
+     */
+    public MqttException(int reasonCode) {
+        super();
```

Comment:
Are u sure? Because the base classes (Exception, Throwable) do this as well and in throwable there is actually work done. I thought it is common to do this in exceptions. I will remove this if you are absolutely sure though.